src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -328,9 +348,27 @@ public function getFileSizeFormatted()
      */
     public function getAttachments()
     {
-        foreach ($this->product['attachments'] as &$attachment) {
-            if (!isset($attachment['file_size_formatted'])) {
-                $attachment['file_size_formatted'] = Tools::formatBytes($attachment['file_size'], 2);
+        // If this is a first call to this property
+        if (!isset($this->product['attachments'])) {
+            $this->product['attachments'] = [];
+
+            /*
+             * There is an optional cache_has_attachments property, which if passed, informs us if the product
+             * has attachments or not in a fast way. We will load attachments only if this property was not passed
+             * or is true.
+             *
+             * This property which needs to be managed every time a file is changed.
+             * It can sometimes lead to database inconsistency.
+             */
+            if (!isset($this->product['cache_has_attachments']) || $this->product['cache_has_attachments']) {
```

Comment:
```suggestion
            if (!empty($this->product['cache_has_attachments'])) {
```

Reply:
This would change the behavior, that's why I kept the current logic.

`if (!isset($this->product['cache_has_attachments']) || $this->product['cache_has_attachments']) {`
This means `if we don't know (it wasn't passed)` OR `we know` that product has attributes, load them.

`if (!empty($this->product['cache_has_attachments'])) {`
This means `if we know`, that product has attributes, load them.

I want to make it as easy to use as possible.